pyracf/common/security_admin.py

```diff
@@ -79,11 +83,28 @@ def __init__(
             self.__replace_valid_segment_traits(replace_existing_segment_traits)
         if additional_secret_traits is not None:
             self.__add_additional_secret_traits(additional_secret_traits)
+        self.set_running_userid(run_as_userid)
+
+    # ============================================================================
+    # Run as Other User ID
+    # ============================================================================
+    def set_running_userid(self, userid: Union[str, None]) -> None:
+        if userid is None:
+            self.__running_userid = None
+            return
+        if not isinstance(userid, str) or len(userid) > 8 or userid == "":
+            raise UserIdError(userid)
+        self.__running_userid = userid.upper()
```

Comment:
Do we want to convert the userid to lowercase here instead of in `get_running_userid()`?

If we do that, then `get_running_userid()` is just this:
```python
    def get_running_userid(self) -> Union[str, None]:
        return self.__running_userid
```
Also need to specify that the return value is `Union[str,None]`.

Reply:
Change made to Union[str, None] in type hints. Unfortunately, while this CAN be done, I think this is the best case scenario. After testing, it seems like we would NEED to ensure the "__running_userid" value is uppercase before passing it to RACF, so checks to drive the string to upper and lowercase will have to go somewhere, and validity checks to ensure that the values ARE strings that can be driven to upper or lower case are also necessary. I would rather have all of this done ONLY when the values are changed or grabbed rather than for every request made to RACF, but I am open to further discussion.

Reply:
That is fine. I think we can revisit this later since this was pretty minor. Let's get this release out.